biorbd_optim/constraints.py

```diff
@@ -104,11 +104,26 @@ def __markers_to_pair(ocp, nlp, X, policy):
 
     @staticmethod
     def __proportional_variable(ocp, nlp, V, policy):
-        for v in V:
-            v = nlp["dof_mapping"].expand(v)
-            ocp.g = vertcat(ocp.g, v[policy[0]] - policy[2] * v[policy[1]])
-            ocp.g_bounds.min.append(0)
-            ocp.g_bounds.max.append(0)
+        """
+        Adds proportionality constraint between the elements (states or controls) chosen.
+        :param nlp: An instance of the OptimalControlProgram class.
+        :param V: List of states or controls at instants on which this constraint must be applied.
+        :param policy: A tuple or a tuple of tuples whose first two elements are the indexes of elements to be linked proportionally.
+        The third element of each tuple (policy[i][2]) is the proportionality coefficient.
+        """
+        if isinstance(policy[0], tuple):
+            for elem in policy:
+                for v in V:
+                    v = nlp["dof_mapping"].expand(v)
+                    ocp.g = vertcat(ocp.g, v[elem[0]] - elem[2] * v[elem[1]])
+                    ocp.g_bounds.min.append(0)
+                    ocp.g_bounds.max.append(0)
```

Comment:
Do not duplicate the code, put the policy in a list if not tuple and reuse the code

Reply:
Also test isinstance(policy[0], (tuple, list))

Reply:
Code condensed + now tests tuple and lists + raises error if there is a unusable mix of tuples/lists and non tuples/lists in policy.

Reply:
[EDIT] I left a mistake in this, fixed it in the commit 6b3626f97ad5b278a6471610d51d2e2b325b671a